packages/contracts-bedrock/src/L2/SuperchainERC20.sol

```diff
@@ -0,0 +1,140 @@
+
+    /// @notice Emitted whenever tokens are successfully relayed on this chain.
+    /// @param to     Address of the recipient.
+    /// @param amount Amount of tokens relayed.
+    /// @param data   Data sent with the message.
+    event RelayedERC20(address indexed to, uint256 amount, bytes data);
+
+    /// @notice A modifier that only allows the bridge to call
+    modifier onlyBridge() {
+        if (msg.sender != BRIDGE) revert CallerNotBridge();
+        _;
+    }
+
+    /// @notice Semantic version.
+    /// @custom:semver 1.0.0
+    string public constant version = "1.0.0";
+
+    /// @param _name        ERC20 name.
+    /// @param _symbol      ERC20 symbol.
+    /// @param _decimals    ERC20 decimals.
+    constructor(string memory _name, string memory _symbol, uint8 _decimals) ERC20(_name, _symbol) {
+        DECIMALS = _decimals;
+    }
+
+    /// @notice Allows the StandardBridge to mint tokens.
+    /// @param _to     Address to mint tokens to.
+    /// @param _amount Amount of tokens to mint.
+    function mint(address _to, uint256 _amount) external virtual override onlyBridge {
+        _mint(_to, _amount);
+        emit Mint(_to, _amount);
+    }
+
+    /// @notice Allows the StandardBridge to burn tokens.
+    /// @param _from   Address to burn tokens from.
+    /// @param _amount Amount of tokens to burn.
+    function burn(address _from, uint256 _amount) external virtual override onlyBridge {
+        _burn(_from, _amount);
+        emit Burn(_from, _amount);
+    }
+
+    /// @notice Sends tokens to some target address on another chain.
+    /// @param _to      Address to send tokens to.
+    /// @param _amount  Amount of tokens to send.
+    /// @param _chainId Chain ID of the destination chain.
+    /// @param _data    Data to be sent with the message.
+    function sendERC20(address _to, uint256 _amount, uint256 _chainId, bytes memory _data) external override {
+        _burn(msg.sender, _amount);
+
+        bytes memory _message = abi.encodeCall(this.relayERC20, (_to, _amount, _data));
+        IL2ToL2CrossDomainMessenger(MESSENGER).sendMessage(_chainId, address(this), _message);
+
+        emit SentERC20(msg.sender, _to, _amount, _chainId, _data);
+    }
+
+    /// @notice Relays tokens received from another chain.
+    /// @param _to     Address to relay tokens to.
+    /// @param _amount Amount of tokens to relay.
+    /// @param _data   Data sent with the message.
+    function relayERC20(address _to, uint256 _amount, bytes memory _data) external override {
+        if (msg.sender != MESSENGER) revert RelayMessageCallerNotL2ToL2CrossDomainMessenger();
+
+        if (IL2ToL2CrossDomainMessenger(MESSENGER).crossDomainMessageSender() != address(this)) {
+            revert MessageSenderNotThisSuperchainERC20();
+        }
+
+        _mint(_to, _amount);
+
+        if (_data.length > 0) {
+            bool success = SafeCall.call(_to, 0, _data);
```

Comment:
@0xng Just being curious:
If this call fails then the tx will be unreliable. Is OP ok with that? Since there is no 'rollback' way at the moment, seems a bit risky to me.
But maybe there are ongoing discussions such us the rollback feature we proposed

Reply:
Actually we pushed a PR today to remove the `_data` parameter and we have mark's approval, the moment it gets merged, we can remove this. But to answer your question, yes, you're right, and yes, OP is ok with that for now

Reply:
Awesome, good to know!